File: code/data_storage/python/access/redis_data_access.py
# SPDX-License-Identifier: GPL-3.0-or-later
# Copyright (C) 2025-2026 ggeoffre, LLC

from sensor_data_access_protocol import SensorDataAccess


class RedisDataAccess(SensorDataAccess):
    """Redis implementation of SensorDataAccess."""

    def log_sensor_data(self, json_data: str) -> None:
        print(f"Logging sensor data to Redis: {json_data}")

    def fetch_sensor_data(self) -> list[str]:
        print("Fetching sensor data from Redis")
        return ['{"sensor": "temperature", "value": 22.3}']

    def purge_sensor_data(self) -> None:
        print("Purging sensor data from Redis")

File: code/data_storage/java/access/src/main/java/com/ggeoffre/data/access/RedisDataAccess.java
// SPDX-License-Identifier: GPL-3.0-or-later
// Copyright (C) 2025-2026 ggeoffre, LLC

package com.geoffrey.data.access;

import java.util.Collections;
import java.util.List;

public class RedisDataAccess implements SensorDataAccess {

    @Override
    public void logSensorData(String jsonData) {
        System.out.println("Logging sensor data to Redis: " + jsonData);
    }

    @Override
    public List<String> fetchSensorData() {
        System.out.println("Fetching sensor data from Redis");
        return Collections.singletonList(
            "{\"sensor\": \"temperature\", \"value\": 22.3}"
        );
    }

    @Override
    public void purgeSensorData() {
        System.out.println("Purging sensor data from Redis");
    }
}
